File: LeetCode75/Middle_of_the_Linked_List/Middle_of_the_Linked_List.cc
// 876. Middle of the Linked List C++


// Definition for singly-linked list.
struct ListNode {
     int val;
     ListNode *next;
     ListNode() : val(0), next(nullptr) {}
     ListNode(int x) : val(x), next(nullptr) {}
     ListNode(int x, ListNode *next) : val(x), next(next) {}
 };

class Solution {
public:
    ListNode* middleNode(ListNode* head) {
        int count = 1;
        ListNode* ans = head;
        while (head) {
            if (count == 2) {
                ans = ans->next;
                count = 0;
            }
            count += 1;
            head = head->next;
        }
        return ans;
    }
};

File: LeetCode75_LevelOne/Middle_of_the_Linked_List/MIddle_of_the_Linked_List.py
# 876. Middle of the Linked List Python
# Definition for singly-linked list.
class ListNode(object):
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution(object):
    def middleNode(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        ans = head
        count = 1
        while head:
            if count == 2:
                ans = ans.next
                count = 0
            count += 1
            head = head.next
        return ans
        